src/Undefinable/map.ts

```diff
@@ -11,7 +11,7 @@ import { Undefinable } from './Undefinable';
  *      * If you'd like return `Undefinable<*>` as `U`, use `andThen()`.
  *      * If the result of _selector_ is `undefined`, this throw an `Error`.
  */
-export function mapForUndefinable<T, U>(src: Undefinable<T>, selector: MapFn<T, U>): Undefinable<U> {
+export function mapForUndefinable<T, U>(src: Undefinable<T>, selector: MapFn<T, NotUndefined<U>>): Undefinable<U> {
```

Comment:
I suspect that it might be less valuable to change `MapFn<T, U>` to `MapFn<NotUndefined<T>, NotUndefined<U>>`.  In this case, the importance is an output value to prevent misusing and it might be meaningless to ban a function which expects that the input is null value.